lang/java/avro/src/main/java/org/apache/avro/util/RandomData.java

```diff
@@ -146,6 +150,23 @@ private Object generate(Schema schema, Random random, int d) {
 
   private static final Charset UTF8 = StandardCharsets.UTF_8;
 
+  private int randomInt(Random random, LogicalType type) {
+    if (type instanceof LogicalTypes.TimeMillis) {
+      return random.nextInt((int) Duration.ofDays(1).toMillis() - 1);
+    }
+    // LogicalTypes.Date LocalDate.MAX.toEpochDay() > Integer.MAX;
+    return random.nextInt();
+  }
+
+  private long randomLong(Random random, LogicalType type) {
+    if (type instanceof LogicalTypes.TimeMicros) {
+      return ThreadLocalRandom.current().nextLong(Duration.ofDays(1).toMillis() * 1000L);
```

Comment:
`Duration.ofDays(1).toMillis() * 1000L` is `86400000000`. IMO it would be good to make it a proper Java constant (`private static final int`).

Reply:
You could still use `Random#nextInt(86400000000)` and cast it to `long`.

Reply:
constant reused